src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -353,6 +353,81 @@ To reiterate how the script system loads script files
+
+If you need to add a new library of program to the Docker image you should add it to the end of the Dockerfile to minimise the amount of rebuilding that needs to happen. Look for the following marker
+
+```sh
+#######################################
+### ADD NEW PROGRAMS/LIBRARIES HERE ###
+#######################################
+```
+
+### Adding a new library to the robot toolchain
+
+If the intention is to have the new library running on the robot then you need to build the library from source. If the library you are building is well-behaved then it should be a simple matter of add a line like the following to the Dockerfile
+
+```docker
+RUN install-from-source http://url.to.the.source.tarball.com
+```
+
+---
+
+**SPOILER**
+Libraries are rarely ever well-behaved
+
+---
+
+This process generally requires so studying of either the libraries documentation or build system files to determine appropriate configure and build flags.
+If there are extra flags that need to be specified then modify the above command to
+
+```docker
+RUN install-from-source http://url.to.the.source.tarball.com \
+    -DCONFIGURE_FLAG=XXX
+```
+
+---
+
+**NOTE**
+`\` is a line continuation operator, it allows a long line to be split over multiple lines to improve readability
+
+---
+
+If your library is not well-behaved you may need to do to a lot of googling and trawling through stackoverflow, bug reports, or github issue pages in order to find workarounds or patches to help you build your library.
+Another good source of information is the `PKGBUILD` scripts from the Arch Linux repositories. [Search](https://www.archlinux.org/packages/) for the name of the library, the look for and click the `Source Files` link on the package page and then click on `PKGBUILD`.
+The `PKGBUILD` script will show you how Arch Linux builds the library including
+
+- necessary dependendies,
+- locations of source files and patches,
+- commands needed to prepare the source files for compilation,
+- commands needed to build the library, and
+- commands needed to install the library
+
+Borrowing commands from these scripts could help relieve your frustrations with the world.
+
+### Adding a new build utility
+
+If the library/program is not intended to be used on the robot, but is instead intended to run inside the Docker image as an extra utility to help build the NUbots code then your first port of call should be the Arch Linux packages. Search for a [package here](https://www.archlinux.org/packages/) (or google "Arch Linux" and the name of the program/library) and if you meet with success add the following command to the Dockerfile
```

Comment:
Specifically, try to add it to the block already at the end I suppose

Reply:
Its a minor detail, easier to just add it at the marker (easier to describe), we can handle clean ups at a later date